### PR TITLE
1837: Fixed financial pre-4-train omissions and bugs

### LINKED ISSUE #359: 1835: Layout ISR window is weird
Rails version 2.4.5, 1835, Clemens variant.

The layout of the ISR window is weird. In earlier versions, the available certificates were just listed top to bottom. In this version there is first a list of four certs, then next to it a list of three certs, etc. 
Is this something intended, or did I do something wrong?

### PR DESCRIPTION
**Pull request by erik-vos**
Bug fixes:

- Enabled 'Split' option for major company revenue
- Fixed rounding issues
- Fixed price dropping on selling more than one share
- Implemented max. 2 companies per par price
- Fixed price changes of sold-out majors
- Fixed incorrect standard mine revenues

Improvements:

- Generic rounding methods, also applied to 18EU
- Implemented the remaining aspects of the Romoth variant

Closes #359, #361, #372, #373, #374, #375, #376

### CODE REVIEW

**Comment by erik-vos** [comment]
All installer tests of this PR fail, so something must be wrong. I cannot research that, but I got a glimpse of what looked like a problem with Java 15. Is that update incomplete, or bad, or what? Can we revert?

**Comment by neutronc** [comment]
The installer tests were successfull for that commit. So something changed with your PRs need to investigate what happened there.

**Comment by erik-vos** [comment]
Some regression tests in the master branch currently fail. Do the installation tests repeat these tests? If so, the  cause is clear.

**Comment by erik-vos** [comment]
OK, I have removed the 4 failing saved test files and their reports.

Two of these are 1837 saved files that were known to be failing and will have to be created anyway
The other two were:
- 18Scan_AfterBankruptcy1.rails 
- 1880_ATG1
These failures is very strange, because AFAIK nothing significant has been changed in the generic code.

In the 18Scan case, the problem seems to be, that a company gets a turn although it should not. It is 'hibernating' (i.e. inactive) because its president has gone bankrupt.

I have not looked at the 1880 case; I will check it, but I know very little about this game, so maybe I will have to delegate this one.

Anyway, all installer tests now succeed, so I suppose this PR is ready to be merged.

File: src/main/java/net/sf/rails/game/OperatingRound.java [hunk 1]
```diff
@@ -2757,6 +2767,106 @@ public void splitRevenue(int amount) {
 
     }
 
+    /*
+     * =======================================
+     *  6.4. ROUNDING
```
Comment (neutronc) [inline src/main/java/net/sf/rails/game/OperatingRound.java:2772]: Good Idea

**Review by neutronc — APPROVED** [review]

**Comment by erik-vos** [comment]
After a long day of debugging I found the small error that inhibited an 18Scan company to be restarted after its presidency got into the Pool after a bankruptcy. Miraculously, this also fixed the 1880 saved file problem.
Also StockRound has been updated to add an overlooked option to restart such a company.
I'll make a new PR for this one.

**Comment by madoar** [comment]
All GitHub Actions pipelines call `./gradlew clean test`. Most pipelines also call `./gradlew clean build`, which also contains a `./gradlew test` call. I'm not entirely sure how the tests for Rails are configured but I imagine they are called entirely

**Pull request merged**